### DISCUSSION TITLE
Underlined `_` cannot be rendered

### DISCUSSION THREAD

**Discussion by RanX12** [KaTeX/KaTeX#2622] [Q&A]
First, I write a mathematical formula like this:
`$$ \sideset{^1_2}{^3_4}\bigotimes \quad or \quad {^1_2}\bigotimes {^3_4} $$`

or
`${x}_{2}$,${x}_{2}$` The following error will appear:
![image](https://user-images.githubusercontent.com/46048444/98100333-acba0a80-1ecb-11eb-8ffe-3eb87164ded8.png)

my code:
![image](https://user-images.githubusercontent.com/46048444/98100476-dd9a3f80-1ecb-11eb-821d-60396b174c28.png)

**Comment by edemaine** [comment]
The error message you include makes it clear that your parser is passing `{x}_{2}$,${x}_{2}` into KaTeX, but it should be passing in `{x}_{2}` and `{x}_{2}` separately.

I think you need to change your regular expressions in `mathsExpression` to use `*?` instead of `*`. Specifically, replace each `[\s\S]*` with `[\s\S]*?` (or the equivalent `[^]*?`).

**Comment by RanX12** [comment]
`{x}_{2}` and `{x}_{2}` is ok.
But I don't quite understand this, can you explain it with specific code? Thank you~

![image](https://user-images.githubusercontent.com/46048444/98190832-e41fca00-1f52-11eb-93a4-43f167b721c5.png)

**Comment by edemaine** [comment]
I'm glad the regular expression fix worked! (if I understand correctly)

Unfortunately, `\sideset` is not supported by KaTeX. See https://katex.org/docs/support_table.html

Something like `{}^1_2\bigotimes{}^3_4` would recreate a similar format, I guess, but I don't know the specific spacing of `\sideset` which is likely better.

**Comment by RanX12** [comment]
ok，I solved this problem, thank you, but I have a new problem. I don’t know if it’s yours. I rendered the formula on the bootstrap popover and the style became very ugly.
![image](https://user-images.githubusercontent.com/46048444/98343361-aef7a280-204c-11eb-948d-8d87c2a6c7b6.png)

If you know how to deal with it, please tell me, thx

**Comment by edemaine** [comment]
This is likely caused by some CSS conflict. We'd need to see a full example page to help diagnose. Or you can inspect the elements and look for  KaTeX CSS rule that's overridden by a Bootstrap CSS rule.
